vector/src/main/java/im/vector/app/features/home/room/detail/timeline/format/NoticeEventFormatter.kt

```diff
@@ -465,21 +465,76 @@ class NoticeEventFormatter @Inject constructor(
 
     private fun formatRoomCanonicalAliasEvent(event: Event, senderName: String?): String? {
         val eventContent: RoomCanonicalAliasContent? = event.getClearContent().toModel()
-        val canonicalAlias = eventContent?.canonicalAlias
-        return canonicalAlias
-                ?.takeIf { it.isNotBlank() }
-                ?.let {
-                    if (event.isSentByCurrentUser()) {
-                        sp.getString(R.string.notice_room_canonical_alias_set_by_you, it)
-                    } else {
-                        sp.getString(R.string.notice_room_canonical_alias_set, senderName, it)
-                    }
+        val prevContent: RoomCanonicalAliasContent? = event.resolvedPrevContent().toModel()
+        val canonicalAlias = eventContent?.canonicalAlias?.takeIf { it.isNotEmpty() }
+        val prevCanonicalAlias = prevContent?.canonicalAlias?.takeIf { it.isNotEmpty() }
+        val altAliases = eventContent?.alternativeAliases.orEmpty()
+        val prevAltAliases = prevContent?.alternativeAliases.orEmpty()
+        val added = altAliases - prevAltAliases
+        val removed = prevAltAliases - altAliases
+
+        return if (added.isEmpty() && removed.isEmpty() && canonicalAlias == prevCanonicalAlias) {
```

Comment:
I converted the top-level `if/else` branches to the `when` expression and it seems more readable to me, especially for the sake of comments (after auto converting to "when", we may move comments one line above).

Reply:
Ok done, please review the commit :)